internal/monitor/otelexporters.go

```diff
@@ -96,8 +99,45 @@ func setupCloudMonitoring(secs int64) []metric.Option {
 		return nil
 	}
 
-	r := metric.NewPeriodicReader(exporter, metric.WithInterval(time.Duration(secs)*time.Second))
-	return []metric.Option{metric.WithReader(r)}
+	// Wrap the exporter to handle permission denied errors
+	wrappedExporter := &permissionAwareExporter{
+		Exporter: exporter,
+	}
+
+	reader := metric.NewPeriodicReader(wrappedExporter, metric.WithInterval(time.Duration(secs)*time.Second))
+	return []metric.Option{metric.WithReader(reader)}
+}
+
+// permissionAwareExporter wraps a metric.Exporter and disables itself if it encounters
+// a PermissionDenied error. This prevents log spam when the environment lacks
+// necessary permissions.
+type permissionAwareExporter struct {
+	metric.Exporter
+	// disabled indicates whether the exporter has been permanently disabled.
+	disabled atomic.Bool
+}
+
+func (e *permissionAwareExporter) Export(ctx context.Context, rm *metricdata.ResourceMetrics) error {
+	// Check if disabled before attempting export to save resources and avoid noise.
+	if e.disabled.Load() {
```

Comment:
do we really need a wrapper here or just incorporate this one condition into existing exporter itself? Because of all these wrappers we are seeing go routine stack size increasing which triggers stack.resize consuming cpu cycles